src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -539,12 +540,102 @@ void Subgraph::createPrimitive() {
         initPluginBlockedShapes();
         initAttributes();
         initStartOffsets();
+        initBroadcastableInputs();
         optimizeIR();
     }
 
     Node::createPrimitive();
 }
 
+void Subgraph::initBroadcastableInputs() {
+    // Snippets supports tokenization of the following operations:
+    // - Unary, Binary and Ternary (Select) Elementwise ops
+    // - Softmax, MatMul, Transpose, GroupNorm
+    // Binary Elementwise ops (+ Select) requires explicit Broadcast op
+    // on inputs if broadcasting of latest dimensions is needed.
+    // These ops will be start points of DFS - need to go to Parameters and update `broadcastable_inputs_map`.
+    // We iterates through all ops by execution order. So if we already analyzied some op in the input branch - skip this branch.
+    // However, there some ops which can change `processing_dim_idx`:
+    // - Transpose has order which changes `processing_dim_idx`. But Transpose can be only before Parameters
+    // - MatMul's first input doesn't affect output latest dimension - skip this branch.
+    //   Also MatMul has `transposed_b` which changes `processing_dim_idx`
+    broadcastable_inputs.clear();
+    const auto& body = subgraph_attrs->snippet->body_ptr();
+    if (body->get_parameters().size() < 2)
+        return;
+
+    const auto& ops = body->get_ordered_ops();
+    std::set<std::shared_ptr<ov::Node>> visited_ops = {};
+    for (const auto& op : ops) {
+        if (!ov::snippets::lowered::pass::InsertBroadcastMove::is_broadcasting_supported(op))
+            continue;
+
+        size_t processing_dim_idx = 0;
+
+        // DFS
+        std::stack<std::shared_ptr<ov::Node>> nodes_to_calculate;
```

Comment:
Please consider using `ov::op::util::visit_path` for graph traversal. From the first glance, it seems that it can be reused here

Reply:
Discussed offline: we have to use DFS here with stack to propagate `processing_dim_idx` from input of the target node to the parameter while `ov::op::util::visit_path` uses BFS with deque.
Added comment with explanation.

Reply:
Thanks!